### ISSUE TITLE
Setting Filename as PDF Header

### ISSUE THREAD

**Issue by vkb008** [coherentgraphics/python-libcpdf#7]
After trying this code (trying to define a header in my pdf files which is the filename of the pdf):

```
    activityPdf = pycpdflib.fromFile(filename,'')
    r = pycpdflib.all(activityPdf)
    activityPdfWithHeader = pycpdflib.addTextSimple(activityPdf,r,'%filename',(pycpdflib.posCentre,20.0,20.0),pycpdflib.timesRoman,6.0)
    pycpdflib.toFile(activityPdfWithHeader,filename,False,False)
```

I get the following error:

`AttributeError: 'NoneType' object has no attribute 'pdf'`

Any suggestions? It appears my implementation of `pycpdflib.addTextSimple` implementation isn't correct.

**Comment by johnwhitington** [comment]
Have you commented lines out to work out which line the problem is at?

**Comment by johnwhitington** [comment]
Ok, the problem here is that addTextSimple doesn't return a new PDF - it operates on the existing PDF.

So you want:

```
activityPdf = pycpdflib.fromFile(filename,'')
    r = pycpdflib.all(activityPdf)
    pycpdflib.addTextSimple(activityPdf,r,'%filename',(pycpdflib.posCentre,20.0,20.0),pycpdflib.timesRoman,6.0)
    pycpdflib.toFile(activityPdf,filename,False,False)
```

Though you probably want to pick a different filename for writing.

**Comment by vkb008** [comment]
@johnwhitington thank you for the suggestion! That cleared the error.

However the addTextSimple method did not add any header text to the pdf. Is the implementation of this method not correct?

I tried this code:
`    pycpdflib.addTextSimple(activityPdf,r,'test',(pycpdflib.posCentre,20.0,20.0),pycpdflib.timesRoman,50.0)`

And this was the output:

![image](https://github.com/coherentgraphics/python-libcpdf/assets/4166458/795eaa4e-b15c-45e0-b730-da6e9f0e636b)

**Comment by vkb008** [comment]
@johnwhitington 

Ok so i tried the following and i got the 'test' string to appear as the header using the following:
`    pycpdflib.addTextSimple(activityPdf,r,'test',(pycpdflib.top,20.0,20.0),pycpdflib.timesRoman,6.0)
`

However, when I tried:
`    pycpdflib.addTextSimple(activityPdf,r,'%filename',(pycpdflib.top,20.0,20.0),pycpdflib.timesRoman,6.0)
`
Header disappeared. I believe I'm using the %filename implementation incorrectly.

To confirm my suspicion, I used a variable that is storing the filename ('xxxxx_xxxxx_xxxx_xxxxx.pdf'):
`    pycpdflib.addTextSimple(activityPdf,r,filename,(pycpdflib.top,20.0,20.0),pycpdflib.timesRoman,6.0)
`

Looks like this worked.

**Comment by johnwhitington** [comment]
The %filename special is only for use with the command line product. In the Python library, the filename is not stored. I will make sure to document this for the next release.

**Comment by vkb008** [comment]
@johnwhitington ah ok, that makes sense why using that command didn't work in my python script. You may also want to specify this detail in your pycpflib.py file because when I saw this comment in addText:

![image](https://github.com/coherentgraphics/python-libcpdf/assets/4166458/b55bc3a8-7530-4961-8913-f0ea8b54e8d8)

It seemed to imply special codes could be used as an input. A coding rookie like me can be confused very easily :)

Another thought - if you could provide additional simple examples of how to implement a few important methods (addText, addTextSimple, adding header/footers/images, etc.), that might help users as well.

That set aside, your library does work great - nice job!

**Comment by johnwhitington** [comment]
Documentation fixed for v2.7 in C and Python products. Thanks.
